lib/middleware/html_response_transformer.rb

```diff
@@ -0,0 +1,40 @@
+require "next_gen_images"
+require "lazy_load_images"
+require "responsive_images"
+require "external_links"
+
+module Middleware
+  class HtmlResponseTransformer
+    def initialize(app)
+      @app = app
+    end
+
+    def transform_response?(headers, response)
+      headers["Content-Type"]&.include?("text/html") &&
+        response.is_a?(ActionDispatch::Response::RackBody)
+    end
+
+    def call(env)
+      status, headers, response = @app.call(env)
+
+      if transform_response?(headers, response)
+        response = [transform(response.body)]
+        headers["Content-Length"] = response.first.bytesize.to_s
+      end
+
+      [status, headers, response]
+    end
+
+    def transform(response_body)
+      response_body = NextGenImages.new(response_body).html
+      response_body = ResponsiveImages.new(response_body).html
+      response_body = LazyLoadImages.new(response_body).html
+      response_body = ExternalLinks.new(response_body).html
```

Comment:
Totally not one for now but it would be nice if this could be chained

```
response_body
  .with_next_gen_images
  .with_responsive_images
  .with_lazy_loading_images
  .with_accessible_external_links
  .without_closing_source_tags
  .html
```

Reply:
Yeah that's a nice suggestion, maybe with a decorator so we don't need to monkey patch the response 🤔